cluster-autoscaler/go.mod

```diff
@@ -10,7 +10,7 @@ require (
 	github.com/Azure/go-autorest/autorest/azure/auth v0.5.8
 	github.com/Azure/go-autorest/autorest/date v0.3.0
 	github.com/Azure/go-autorest/autorest/to v0.4.0
-	github.com/aws/aws-sdk-go v1.38.49
```

Comment:
As we import lots of code from kubernetes/kubernetes want to be in sync with it (k/k uses github.com/aws/aws-sdk-go v1.38.49) to avoid compilation and runtime issues. So we cannot accept any changes that can make future sync problematic. Please update k/k deps or roll back this change.

Reply:
Understood, opened a PR to update dependencies in kubernetes/kubernetes: https://github.com/kubernetes/kubernetes/pull/108897

Reply:
@mwielgus the reviewers on the above PR have some pushback against further changes to `aws-sdk-go` as a legacy cloud provider in k/k: https://github.com/kubernetes/kubernetes/pull/108897#issuecomment-1083689535. What should be the next step here?

For some context, I updated the `github.com/aws/aws-sdk-go` version from v1.38.49 because this integration is dependent on API support for attribute-based instance requirements, introduced in [v1.41.12](https://github.com/aws/aws-sdk-go/releases/tag/v1.41.12).

Reply:
@mwielgus ping :)

Reply:
@feiskyer Possible for you to do look into this ? we need to make progress

Reply:
@gjtempleton may I ask if you have an opinion on where to go from here, with regards to k/k's stance on provider updates and changes like this in Autoscaler?